satellite/console/consoleweb/server.go

```diff
@@ -129,7 +130,17 @@ func (s *Server) Run(ctx context.Context) error {
 		return Error.Wrap(err)
 	}
 
-	return s.server.Serve(s.listener)
+	group, ctx := errgroup.WithContext(ctx)
+	group.Go(func() error {
+		<-ctx.Done()
+		return s.server.Shutdown(nil)
+	})
+
+	group.Go(func() error {
+		return s.server.Serve(s.listener)
+	})
+
+	return group.Wait()
```

Comment:
Maybe similar code from pkg/server/server.go:100 should be adjusted to this form? it looks simpler and cleaner.

Reply:
fixed

Reply:
I just realized that it wouldn't close the other goroutine when only the serving returns.